packages/transformer/src/test/standalone/BranchProvenanceInitializer.test.ts

```diff
@@ -0,0 +1,248 @@
+    model: IModelDb.repositoryModelId,
+    url: args.masterUrl,
+    format: "iModel",
+    repositoryGuid: args.master.iModelId,
+    description: args.masterDescription,
+  }, args.branch).insert();
+
+  const masterExternalSourceId = new ExternalSource({
+    classFullName: ExternalSource.classFullName,
+    model: IModelDb.rootSubjectId,
+    code: Code.createEmpty(),
+    repository: new ExternalSourceIsInRepository(masterLinkRepoId),
+    connectorName: require("../../../../package.json").name,
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    connectorVersion: require("../../../../package.json").version,
+  }, args.branch).insert();
+
+  // initialize the branch provenance
+  const branchInitializer = new IModelTransformer(args.master, args.branch, {
+    // tells the transformer that we have a raw copy of a source and the target should receive
+    // provenance from the source that is necessary for performing synchronizations in the future
+    wasSourceIModelCopiedToTarget: true,
+    // store the synchronization provenance in the scope of our representation of the external source, master
+    targetScopeElementId: masterExternalSourceId,
+  });
+
+  await branchInitializer.processAll();
+  // save+push our changes to whatever hub we're using
+  const description = "initialized branch iModel";
+  args.branch.saveChanges(description);
+
+  branchInitializer.dispose();
+
+  return {
+    masterExternalSourceId,
+    targetScopeElementId: masterExternalSourceId,
+    masterRepositoryLinkId: masterLinkRepoId,
+  };
+}
+
+function setToStandalone(iModelName: string) {
+  const nativeDb = new IModelHost.platform.DgnDb();
+  nativeDb.openIModel(iModelName, OpenMode.ReadWrite);
+  nativeDb.setITwinId(Guid.empty); // empty iTwinId means "standalone"
+  nativeDb.saveChanges(); // save change to iTwinId
+  nativeDb.deleteAllTxns(); // necessary before resetting briefcaseId
+  nativeDb.resetBriefcaseId(BriefcaseIdValue.Unassigned); // standalone iModels should always have BriefcaseId unassigned
+  nativeDb.saveLocalValue("StandaloneEdit", JSON.stringify({ txns: true }));
+  nativeDb.saveChanges(); // save change to briefcaseId
+  nativeDb.closeIModel();
+}
+
+export function generateEmptyIModel(fileName: string): InsertParams {
+  const sourcePath = IModelTransformerTestUtils.prepareOutputFile("IModelTransformer", fileName);
+  if (fs.existsSync(sourcePath))
+    fs.unlinkSync(sourcePath);
+
+  const db = StandaloneDb.createEmpty(sourcePath, { rootSubject: { name: fileName }});
+
+  const physModelId = PhysicalModel.insert(db, IModelDb.rootSubjectId, "physical model");
+  const categoryId = SpatialCategory.insert(db, IModelDb.dictionaryId, "spatial category", {});
+
+  return {
+    db,
+    physModelId,
+    categoryId,
+  };
+}
+
+function insertElementToImodel(insertData: InsertParams, withFedGuid: boolean, index: number): string {
```

Comment:
inline this and instead extract common props out:
```ts
const baseProps = {
  geom: blah,
};

const elem = new PhysicalObject({
  ...baseProps,
  userLabel: "blah",
}, db).insert();
```
etc

Reply:
done